GpsDataAnalyzer/calculator/alignment_algorithms.py

```diff
@@ -0,0 +1,279 @@
+
+def find_lineup_no_optimization(set1, set2):
+  """
+  Find the approximate offset between two GPS data sets without range start optimization.
+
+  This algorithm first identifies a primary data set and a secondary one based
+  on which starts later. The offset is then applied to the secondary data set.
+  After mapping the timestamps to the corresponding points from each set,
+  it finds the optimal offset using the later start time of the two sets as
+  the starting point. Requires checking of more values in the set to obtain
+  an accurate offset, compared to the optimized version.
+
+  Args:
+    set1: GpsDataSet object
+    set2: GpsDataSet object
+
+  Returns:
+    Tuple of two datetimes, the start time for set 1 and for set 2 given the
+    calculated offset.
+  """
+  set1_start_time = set1.gps_data_list[0].time
+  set2_start_time = set2.gps_data_list[0].time
+  if set1_start_time > set2_start_time:
+    later_start_time = utils.round_time(set1_start_time)
+    primary_set_index = 0
+  else:
+    later_start_time = utils.round_time(set2_start_time)
+    primary_set_index = 1
+
+  # create dict that maps rounded times to points
+  time_point_mapping = create_time_to_point_mapping(set1, set2)
+
+  # find best offset
+  #offset: (-100,100), check points: (-100, 100)
+  optimal_offset = find_optimal_offset(time_point_mapping, later_start_time, 200, 200, primary_set_index)
+  print(optimal_offset)
+  if optimal_offset is None:
+    print("no optimal line-up for these two data sets; check if correct files are being used")
+    return (None, None)
+  if primary_set_index == 0:
+    print("Optimal offset: set 2 is %s seconds from set 1" % optimal_offset)
+    return (later_start_time, later_start_time + timedelta(seconds=optimal_offset))
+  else:
+    print("Optimal offset: set 1 is %s seconds from set 2" % optimal_offset)
+    return (later_start_time + timedelta(seconds=optimal_offset), later_start_time)
+
+def find_lineup_naive(set1, set2):
+  """
+  Initial naive implementation to find lineup between two data sets.
+
+  This algorithm first identifies a primary data set and a secondary one based
+  on which starts later. The offset is then applied to the secondary data set.
+  After mapping the timestamps to the corresponding points from each set, it
+  finds the range that has the least amount of invalid timestamps (has points
+  from only one set or neither) and then finds the optimal offset using the
+  middle of that range as the starting point.
+
+  Args:
+    set1: GpsDataSet object
+    set2: GpsDataSet object
+
+  Returns:
+    The indexes of the starting points in each data set
+  """
+  sets = [set1, set2]
+  starting_indexes = [0,0]
+  set1_start_time = set1.gps_data_list[0].time
+  set2_start_time = set2.gps_data_list[0].time
+  if set1_start_time > set2_start_time:
+    later_start_time = set1_start_time
+    primary_set_index = 0
+    secondary_set_index = 1
```

Comment:
Can we assign `(primary|secondary)_set` instead of referring to them from in an array? It saves a couple of variables and it might make it easier to read.